core/services/ocr2/plugins/ocr2vrf/coordinator/coordinator.go

```diff
@@ -418,14 +511,16 @@ func (c *coordinator) filterEligibleCallbacks(
 			continue
 		}
 
-		// check that the callback hasn't been scheduled for transmission
-		// so we don't fulfill the callback twice.
-		_, transmitted := c.toBeTransmittedCallbacks[callback{
-			blockNumber: r.NextBeaconOutputHeight,
-			requestID:   r.Callback.RequestID.Uint64(),
-			blockHash:   blockhashesMapping[r.NextBeaconOutputHeight],
-		}]
-		if !transmitted && isBlockEligible(r.NextBeaconOutputHeight, r.ConfDelay, currentHeight) {
+		// Check that the callback is elligible, and that it hasn't been scheduled for a valid transmission.
+		// A re-orged transmission can be ignored.
+		if isBlockEligible(r.NextBeaconOutputHeight, r.ConfDelay, currentHeight) {
+			cacheKey := getCallbackCacheKey(r.Callback.RequestID.Int64())
+			t := c.toBeTransmittedCallbacks.GetItem(cacheKey)
+			validTransmission := (t != nil) && (t.recentBlockHash == blockhashesMapping[t.recentBlockHeight])
```

Comment:
does `validTransmission` mean that the callback is in-flight? Maybe `inflightTransmission` is a better name here?

Reply:
yeah it means it is either in-flight or successfully on-chain. I can make that change.